### PR TITLE
testing sycl targets

### PR DESCRIPTION
**Pull request by Ashutosh-Londhe**

### CODE REVIEW

File: makefiles/Makefile.common [hunk 1]
```diff
@@ -30,13 +30,55 @@ OPS_LIB_OMPOFFLOAD=-lops_ompoffload
 OPS_LINK=
 CUDART=-lcudart
 
-COMPILER_AVAILABLE := clang gnu intel icx intel_win mic pgi cray xl hip xlp8 aocc
```
Comment (reguly) [inline makefiles/Makefile.common:33]: I think this was a better way of doing things. Why go back to the old if-else structure?

File: ops/c/src/mpi/ops_mpi_rt_support_sycl.cpp [hunk 2]
```diff
@@ -618,9 +618,9 @@ void ops_dat_fetch_data_slab_memspace(ops_dat dat, int part, char *data, int *ra
       prod *= target->size[d];
     }
     ops_kernel_descriptor *desc = ops_dat_deep_copy_core(target, dat, range);
-    desc->name = "ops_internal_copy_device";
-    desc->device = 1;
-    desc->function = ops_internal_copy_device;
+    strcpy(desc->name, "ops_internal_copy_device\0");
```
Comment (reguly) [inline ops/c/src/mpi/ops_mpi_rt_support_sycl.cpp:621]: Why do we need the null character here?

**Review by reguly — CHANGES_REQUESTED** [review]

File: makefiles/Makefile.common (hunk 1 above)
Comment (Ashutosh-Londhe) [reply makefiles/Makefile.common:33]: You asked me earlier to revert this

File: ops/c/src/mpi/ops_mpi_rt_support_sycl.cpp (hunk 2 above)
Comment (Ashutosh-Londhe) [reply ops/c/src/mpi/ops_mpi_rt_support_sycl.cpp:621]: name is a char*

Comment (reguly) [reply ops/c/src/mpi/ops_mpi_rt_support_sycl.cpp:621]: yes, but as far as I am aware, when you have a cost char* string, like "ops_internal_copy_device", it implicitly comes with a terminating \0 char at the end. Please check and confirm. If it really does need the \0 at the end, then we'll need that everywhere else in the code too.

Comment (Ashutosh-Londhe) [reply ops/c/src/mpi/ops_mpi_rt_support_sycl.cpp:621]: yes i checked. It is not required.
removing it

File: makefiles/Makefile.common (hunk 1 above)
Comment (reguly) [reply makefiles/Makefile.common:33]: Perhaps I misunderstood which way - but the point is we should have the generic structure which will work with any makefile extension that is present.

**Comment by gihanmudalige** [comment]
@Ashutosh-Londhe is this PR still valid ?

**Pull request closed without merging**